### PR TITLE
Add tests for XEP-0045 Multi-User Chat (section 7.5)

### PR DESCRIPTION
**Pull request by guusdk**

### CODE REVIEW

**Comment by guusdk** [comment]
The test failure highlights a bug in Openfire: https://igniterealtime.atlassian.net/browse/OF-2858

A fix is proposed in https://github.com/igniterealtime/Openfire/pull/2505

File: src/main/java/org/jivesoftware/smackx/muc/MultiUserChatOccupantPMIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+                    if (participant.equals(targetMucAddress)) {
+                        ownerSeesTarget.signal();
+                    }
+                }
+            });
+
+            mucAsSeenByTarget.join(nicknameTarget);
+            ownerSeesTarget.waitForResult(timeout);
+
+            conTwo.addStanzaListener(pmListener, new AndFilter(StanzaTypeFilter.MESSAGE, new StanzaIdFilter(randomString)));
```
Comment (Fishbowler) [inline src/main/java/org/jivesoftware/smackx/muc/MultiUserChatOccupantPMIntegrationTest.java:112]: Does this filter also need to validate that the `from` address matches conOne's occupant JID (as opposed to their real JID)?
I don't think it quite warrants a test of its own.

**Review by Fishbowler — APPROVED** [review]
Aside from my 1 question above about a potential missing check, LGTM!

File: src/main/java/org/jivesoftware/smackx/muc/MultiUserChatOccupantPMIntegrationTest.java (hunk 1 above)
Comment (guusdk) [reply src/main/java/org/jivesoftware/smackx/muc/MultiUserChatOccupantPMIntegrationTest.java:113]: That's a good idea! I've just pushed a new commit that adds that assertion.

**Pull request merged**